arc.go

```diff
@@ -416,3 +416,69 @@ func (al *arcList) RemoveTail() interface{} {
 func (al *arcList) Len() int {
 	return al.l.Len()
 }
+
+//return the ttl of the item
+func (it *arcItem) GetExpired(now *time.Time) (bool, time.Duration) {
+	if it.expiration == nil {
+		return false, time.Second*time.Duration(-1)
+	}
+	if now == nil {
+		t := it.clock.Now()
+		now = &t
+	}
+	return it.expiration.Before(*now), it.expiration.Sub(*now)
+}
+
+//get the TTL(Time to live) of the item
+func (c *ARC) getTTL(key interface{}, onLoad bool) (interface{}, error) {
```

Comment:
If I understand correctly, you want to be able to obtain a cache item's time-to-live. The way it is implemented as of now is not ideal in my opinion because it simulates a `Get` operation and duplicates its logic.

In my opinion, we:

- should not consider observing a key's TTL as a read-access in the classical sense.

- should at least be re-factored so that we get a single piece of logic doing the heavy lifting (membership check, updating priority, updating stats etc.) and two thinner methods on top of it (Getting the item, getting the TTL).

The first point might be subject to debate, I welcome it. IMO, a view of an item's TTL should not be considered as an access to that item because even if in practice it could (in this case *is*, but only because of the way we chose to architecture the cache) it is not from a user perspective. It makes for awkward situations like this:

```
while {
    if cache.GetTTL(k).Seconds() < 10 {
        // do stuff
    }
}
```

Should you count this as 10 read access to `key`? I don't think so.

Here's a possibly easier route we could take to implement the feature you want:
- transparently fetch the TTL (have a method get an item that bypass the cache get logic, it could be used for future development if we enrich the properties attached to an item later on)
- if the item is not found, return a `KeyNotFoundError`
- if the item has no expiration return a `NoExpirationTimeError` or `NonPerishableItem` or whatever name you find fitting. I think that's a better behavior than returning `-1` because it's not obvious from a user perspective that an item with no expiration time will return no error but a negative duration!


`func (c *WhateverCache) GetTTL(key interface{}) (time.Duration, error)`